.github/workflows/tests.yaml

```diff
@@ -44,9 +44,9 @@ jobs:
           key: venv-${{ matrix.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
       - name: Install dependencies
         if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
-        run: poetry install --all-extras --no-interaction --no-root
+        run: poetry install --no-interaction --all-extras --only main,test --no-root
       - name: Install library
-        run: poetry install --all-extras --no-interaction
+        run: poetry install --no-interaction --all-extras --only-root
```

Comment:
Here we only want the package itself to be installed, because we are reusing a cached environment from above. Hence `--only-root` (repeating the same line above without `--no-root` would also work).